### DISCUSSION TITLE
needs_string_links for lists (multiple matches for regex)

### DISCUSSION THREAD

**Discussion by JochenKienzle** [useblocks/sphinx-needs#530] [Ideas]
In my setup I have needs which will reference to multiple external defined tickets e.g. github-links.

The definition would look like:

```
.. need:: Test
  :github-ids: 123, 456, 789
```
I have tried the following regex which should lead to multiple matches:

```
needs_string_links = {
    'github-links': {
        'regex': r'(?P<value>.+?)(?:,|$)',
        'link_url' : 'https://<github-url>{{value}}',
        'link_name': '{{value}}',
        'options': ['github-ids']
    }
}
```

I have use this regex I only get a link created for the first id (in this example: [123](github123)). I assume that only the first match is currently used to generated the link.

If this is the case, I propose to extend it to consider multiple matches and create a list of comma-separated links (in this example: [123](github123), [456](github456), [789](github789)) in the output.

**Comment by danwos** [comment]
That's right, lists are not supported.
But should be easy to implement.

**Reply by rsarrazin2** [reply]
It looks like this feature was implemented in #652. The [latest needs-string-links documentation](https://sphinxcontrib-needs.readthedocs.io/en/latest/configuration.html#needs-string-links) already contains the feature by listing multiple GitHub ids, but not yet the latest release 1.0.1.

**Reply by danwos** [reply]
That's right. It got introduced with Pr #652

**Reply by JochenKienzle** [reply]
I have tested it and it works 👍 

Thank you for the info :)
